app/dao/template_categories_dao.py

```diff
@@ -0,0 +1,30 @@
+import uuid
+
+from sqlalchemy import asc
+
+from app import db
+from app.dao.dao_utils import transactional
+from app.models import TemplateCategory
+
+
+@transactional
+def dao_create_template_category(template_category: TemplateCategory):
+    template_category.id = uuid.uuid4()
+    db.session.add(template_category)
+
+
+@transactional
+def dao_update_template_category(template_category: TemplateCategory):
+    db.session.add(template_category)
+
+
+def dao_get_template_category_by_id(template_category_id):
+    return TemplateCategory.query.filter_by(id=template_category_id).one()
+
+
+def dao_get_template_category_by_template_id(template_id):
+    return TemplateCategory.query.join(TemplateCategory.templates).filter_by(id=template_id).one()
+
+
+def dao_get_all_template_categories():
```

Comment:
lets write this with filters - so if we only want to return sms/ email/ hidden

Reply:
For the SMS/Email filter, are we filtering by template in this case? e.g. `Select all template categories used by at least 1 sms template`

Reply:
I am unsure? I was going to say select all SMS template categories / or select template categories where hidden is not true. something along those lines

Reply:
That's what I had in mind as well, I've implemented filtering so that we can select categories by any combination of `template_type` and `hidden` e.g:

Categories associated with sms templates that are hidden / not hidden
Categories associated with email templates that are hidden / not hidden

I added a lazy join on the category<>template so we can also easily get a list of the templates filtered this way as well if we need to.